### ISSUE TITLE
PartDesign: Additive helix around a cylinder fails with "AdditiveHelix: Tool shape is not valid for boolean operation" unless workaround is employed

### ISSUE THREAD

**Issue by Andrethetech** [FreeCAD/FreeCAD#18072]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Additive helix around a cylinder fails with "AdditiveHelix: Tool shape is not valid for boolean operation" unless I set the Cone angle to 0.01 instead of 0.00 and that is unexpected behavior. I increased the tolerance and it had no effect also turned off refine .

### Full version info

```shell
OS: Windows 11 build 26100
Architecture: x86_64
Version: 1.0.0.39109 (Git) Conda
Build type: Release
Branch: (HEAD detached at 1.0.0)
Hash: 2fcc5317fe3aee96ca73475986a577719fc78e20
Python 3.11.10, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: English/Canada (en_CA)
Stylesheet/Theme/QtStyle: FreeCAD Light.qss/FreeCAD Light/Fusion
```

### Subproject(s) affected?

PartDesign

### Anything else?

![Helix](https://github.com/user-attachments/assets/0b882b1a-d535-4bbf-bd67-f360984476fc)
[leadscrew.zip](https://github.com/user-attachments/files/17872063/leadscrew.zip)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tarman3** [comment]
This file is trying to burn my laptop ))

**Comment by maxwxyz** [comment]
@FlachyJoe @AIRCAP FYI

**Comment by Andrethetech** [comment]
Reduced from 100mm to 50mm , takes about 10-15 sec to compute on my system.
[leadscrew (2).zip](https://github.com/user-attachments/files/17872997/leadscrew.2.zip)

**Comment by tarman3** [comment]
My laptop not so powerful, only 8 gb of RAM and this is not enougth or this file.

**Comment by Andrethetech** [comment]
Ok I will try to reduce it as as I can but since the pitch is 4mm it can't be 5mm , I tried smaller values and I get weird results.

**Comment by tarman3** [comment]
You are wrong.
Length 5 mm. Pitch 10 mm.
![Screenshot_20241122_182102](https://github.com/user-attachments/assets/eb850caa-0d05-438e-bc69-145a112b2231)

**Comment by Andrethetech** [comment]
[leadscrew (3).zip](https://github.com/user-attachments/files/17873262/leadscrew.3.zip)
My bad I was actually capable to redue it to 5mm ,see the attached file

**Comment by tarman3** [comment]
Wow. It is strange that the geometry does not contain errors when constructing a spiral in this way.
Try changing your approach.

![Screenshot_20241122_182736](https://github.com/user-attachments/assets/03df42c9-2204-4880-8fbf-7ff38bfc629f)

**Comment by tarman3** [comment]
To get double thread use **PolarPattern**
![Screenshot_20241122_183534](https://github.com/user-attachments/assets/aa33f552-8487-4ae6-a116-13490515ff9d)

**Comment by Andrethetech** [comment]
I changed the sketch and now it works with the cone angle set to 0.00 and consumes less memory .

**Comment by Andrethetech** [comment]
Yes by changing the sketch.
![Capture d’écran 2024-11-22 114824](https://github.com/user-attachments/assets/7fd406fa-e7e0-4969-af27-3c3fbc5628d6)

**Comment by Andrethetech** [comment]
Yes weird issues with helix and I ran into a issue where I was hitting 100% memory usage and I have 48Gb of ram at some point generating a 500mm leadscrew model.

**Comment by FlachyJoe** [comment]
You can increase Deviation and AngularDeflection view properties of the body in order to save time and resources.

**Comment by maxwxyz** [comment]
Is this still an issue in the latest weekly?

**Comment by FEA-eng** [comment]
The bottom subtractive cylinder leads to something like this:

<img width="487" height="374" alt="Image" src="https://github.com/user-attachments/assets/035b0980-cfd8-43f2-9aeb-63a395f8ceb7" />

```
OS: Windows 11 build 26100
Architecture: x86_64
Version: 1.1.0dev.14555 (Git shallow)
Build date: 2025/09/09 06:43:51
Build type: Release
Branch: main
Hash: c1dd116e1e3304511d9e5e634e711e549b2b84e7
Python 3.11.13, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, IfcOpenShell 0.8.2, OCC 7.8.1
Locale: Polish/Poland (pl_PL)
Navigation Style/Orbit Style/Rotation Mode: Blender/Rounded Arcball/Window center
Stylesheet/Theme/QtStyle: FreeCAD.qss/FreeCAD Light/
Logical DPI/Physical DPI/Pixel Ratio: 96/81.5973/1
```

**Comment by Harvie** [comment]
I am trying to cut simple internal thread, but it fails as well.

[pet_cap.FCStd.zip](https://github.com/user-attachments/files/22729472/pet_cap.FCStd.zip)

Preview is OK tho:

<img width="787" height="620" alt="Image" src="https://github.com/user-attachments/assets/59f63a53-4fe4-4445-96af-c27cc6c8e077" />

Says "warning: helix might be self-intersecting". Is that an issue? i don't see why it should tho...

But even when the helix is not seff-intersecting it either:
  1. fails with: Tool shape is not valid for boolean operation
  2. fails with: Base shape is not valid for boolean operation
  3. does not fail with error message, helix object is created in the tree, but nothing is subtracted (see picture)
 

<img width="739" height="305" alt="Image" src="https://github.com/user-attachments/assets/8067cb0a-d647-470b-bd69-60b170e04298" />


OS: Arch Linux (XFCE/Xfce Session/xcb)
Architecture: x86_64
Version: 1.0.2.39319 (Git) Conda AppImage
Build type: Release
Branch: (HEAD detached at 1.0.2)
Hash: 256fc7eff3379911ab5daf88e10182c509aa8052
Python 3.11.13, Qt 5.15.15, Coin 4.0.3, Vtk 9.3.0, OCC 7.8.1
Locale: Czech/Czech Republic (cs_CZ)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
Installed mods: 
  * sheetmetal 0.7.22
  * Curves 0.6.58

**Comment by Harvie** [comment]
bump

**Comment by FlachyJoe** [comment]
@Harvie self intersecting helix is problematic. And IMHO your design has no sens if the sketch was 3.18mm high and your helix has 3.10mm pitch.

**Comment by FEA-eng** [comment]
Still relevant

```
OS: Windows 10 build 19045
Architecture: x86_64
Version: 1.2.0dev.20260602 (Git shallow)
Build date: 2026/06/02 19:21:10
Build type: Release
Branch: (HEAD detached at 07de249ec)
Hash: 07de249ec749d3fba6710c78da8bca7c66243347
Python 3.11.15, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, Clipper2 , IfcOpenShell 0.8.2, OCC 7.8.1
Locale: Polish/Poland (pl_PL) [ OS: Polish/Poland (pl_PL) ]
Stylesheet/Theme/QtStyle: FreeCAD.qss/FreeCAD Light/
Navigation Style/Orbit Style/Rotation Mode: Touchpad/Rounded Arcball/Window center
Logical DPI/Physical DPI/Pixel Ratio: 96/113.268/1.25
```
